### ISSUE TITLE
Seeking Guidance on Reproducing Reported Accuracy

### ISSUE THREAD

**Issue by aerumnose** [YDU-uva/ProtoDiff#3]
I am currently working on reproducing the reported accuracy of Protonet on miniImageNet with a 1-shot scenario (63.17% -> 66.63%). I initially trained the classifier-baseline using the provided code and attempted to train ProtoDiff.

However, I encountered challenges as the accuracy scores were not as expected. Specifically, after training both the classifier-baseline and ProtoDiff on convnet4 and resnet12, the accuracy scores were suboptimal (resnet12: 63.25% -> 62.88%, convnet4: 41.61% -> 31.52%).

**Do you have plans to release the final code?**

I would appreciate any guidance or insights you can provide on the configurations I used. Please find the details below:

[train_classifier_mini.yaml]
train_dataset: mini-imagenet
train_dataset_args: {split: train, augment: resize}
val_dataset: mini-imagenet
val_dataset_args: {split: train_phase_val}
fs_dataset: mini-imagenet
fs_dataset_args: {split: test}
eval_fs_epoch: 5

model: classifier
model_args:
    encoder: resnet12
    encoder_args: {}
    classifier: linear-classifier 
    classifier_args: {n_classes: 64}

batch_size: 256
max_epoch: 150
optimizer: sgd
optimizer_args: {lr: 0.1, weight_decay: 5.e-4, milestones: [90]}

save_epoch: 5
visualize_datasets: true

[train_meta_mini.yaml]
train_dataset: mini-imagenet
train_dataset_args: { split: train }
tval_dataset: mini-imagenet
tval_dataset_args: { split: test }
val_dataset: mini-imagenet
val_dataset_args: { split: val }

model: meta-baseline
model_args:
  encoder: resnet12
  encoder_args: { }
load_encoder: ./save/classifier_mini-imagenet_resnet12/epoch-last.pth
n_way: 5
n_shot: 1
n_query: 15
train_batches: 200
ep_per_batch: 4

max_epoch: 100
optimizer: sgd
diffusion_optimizer: sgd
optimizer_args: { lr: 1.e-4, weight_decay: 5.e-4}
diffusion_optimizer_args: { lr: 1.e-3, weight_decay: 5.e-4, milestones: [30, 50, 80]}

visualize_datasets: True
overfit_lr: 0.000001
overfit_iterations: 10
loss_coff: 10.

save_epoch: 20

transformer:
  ema: true
  predict_xstart: true
  max_freq_log2: 14
  num_frequencies: 128
  n_embd: 512
  encoder_depth: 1
  decoder_depth: 1
  n_layer: 12
  n_head: 16
  dropout_prob: 0.1
  len_input: 3
  numstep: 10
  time: 100

[test_few_shot.yaml]
dataset: mini-imagenet
dataset_args: {split: test}

load: ./save/meta_mini-imagenet-1shot_meta-baseline-resnet12-1e-06-10/max-va.pth

transformer:
  ema: true
  predict_xstart: true
  max_freq_log2: 14
  num_frequencies: 128
  n_embd: 512
  encoder_depth: 1
  decoder_depth: 1
  n_layer: 12
  n_head: 16
  dropout_prob: 0.1
  len_input: 3
  numstep: 1
  time: 100

visualize_datasets: True
overfit_lr: 0.000001
overfit_iterations: 10
loss_coff: 10.

**Comment by wangpeng0702** [comment]
Hello, did you obtain the 63.17% metric in test_few_stot.cy or train_1. py？

**Comment by ZL-1226** [comment]
The results I reproduced on resnet12, with dataset minimagenet, the accuracy of the diffusion-refactored prototype is consistently lower than the accuracy of the initial prototype with normal training, both during training and testing; I'd like to ask if there are any experimental details that can guide the instruction, or if there is a final code to publish, thank you very much!

**Comment by wangpeng0702** [comment]
> The results I reproduced on resnet12, with dataset minimagenet, the accuracy of the diffusion-refactored prototype is consistently lower than the accuracy of the initial prototype with normal training, both during training and testing; I'd like to ask if there are any experimental details that can guide the instruction, or if there is a final code to publish, thank you very much!

Bro, stop focusing on the experiments in this paper. I’ve been following it since last year, and I’ve come to realize that the author is very likely committing academic fraud—he most likely just reported the results on the validation dataset, and it’s impossible to reproduce the results claimed in the paper.

**Comment by ZL-1226** [comment]
谢谢提醒呀，我就说怎么没效果测试！兄弟你也是做少样本的吗，好奇有在这篇论文基础上做尝试不，我感觉这个框架还是挺不错的可惜了，就是将扩散模型引入原型网络，有空的话可以讨论讨论呀，这就是我的qq



发自我的iPhone


------------------ Original ------------------
From: 汪鹏（wangpeng） ***@***.***&gt;
Date: Sun,Apr 20,2025 10:12 PM
To: YDU-uva/ProtoDiff ***@***.***&gt;
Cc: ZL-1226 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [YDU-uva/ProtoDiff] Seeking Guidance on Reproducing ReportedAccuracy (Issue #3)



    
The results I reproduced on resnet12, with dataset minimagenet, the accuracy of the diffusion-refactored prototype is consistently lower than the accuracy of the initial prototype with normal training, both during training and testing; I'd like to ask if there are any experimental details that can guide the instruction, or if there is a final code to publish, thank you very much!
  
Bro, stop focusing on the experiments in this paper. I’ve been following it since last year, and I’ve come to realize that the author is very likely committing academic fraud—he most likely just reported the results on the validation dataset, and it’s impossible to reproduce the results claimed in the paper.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  wangpeng0702 left a comment (YDU-uva/ProtoDiff#3)
  
The results I reproduced on resnet12, with dataset minimagenet, the accuracy of the diffusion-refactored prototype is consistently lower than the accuracy of the initial prototype with normal training, both during training and testing; I'd like to ask if there are any experimental details that can guide the instruction, or if there is a final code to publish, thank you very much!
  
Bro, stop focusing on the experiments in this paper. I’ve been following it since last year, and I’ve come to realize that the author is very likely committing academic fraud—he most likely just reported the results on the validation dataset, and it’s impossible to reproduce the results claimed in the paper.

**Comment by wangpeng0702** [comment]
> 谢谢提醒呀，我就说怎么没效果测试！兄弟你也是做少样本的吗，好奇有在这篇论文基础上做尝试不，我感觉这个框架还是挺不错的可惜了，就是将扩散模型引入原型网络，有空的话可以讨论讨论呀，这就是我的qq
> 
> 
> 
> 发自我的iPhone
> […](#)

兄弟，qq号被屏蔽了，我这显示不出来

**Comment by yangzq8** [comment]
想问现在复现成功了吗，我也是做小样本的
